installations/src/android/installations_android.cc

```diff
@@ -186,6 +186,19 @@ void InstallationsInternal::Cleanup() {
   }
 }
 
+void InstallationsInternal::LogHeartbeat(const firebase::App& app) {
+  // Calling the native getter is sufficient to cause a Heartbeat to be logged.
+  JNIEnv* env = app.GetJNIEnv();
```

Comment:
I'm curious if there will be some warning and exception thrown out while running on android. If so, probably need to follow this pattern to make sure the jni exception is cleared.

https://github.com/firebase/firebase-cpp-sdk/blob/dcb0b3b3da62b04dc869b0b40508c82238e2516b/remote_config/src/android/remote_config_android.cc#L590

Reply:
Good catch.  We probably should do the same for Auth as well

Reply:
I added a call to CheckAndClearJniExceptions.
I removed the FIREBASE_ASSERT since I had just copied it literally from the above InstallationsInternal, but I dont think we need such an assertion in the LogHeartbeat method since the installations instance is not actually used by LogHeartbeat. 

I also checked the Auth change and confirmed that I do already call CheckAndClearJniExceptions in the Auth LogHeartbeat method.
https://github.com/firebase/firebase-cpp-sdk/pull/846/files

I find it a bit odd that there were not any existing calls to CheckAndClearJniExceptions in installations_android.cc, but maybe it is just because it was written longer ago.